### PR TITLE
(add) new decorator for client_core

### PR DESCRIPTION
**Pull request by agiamas**
To do (delete all that do not apply):

 - [X] Change has a jira ticket that has the correct status.
 - [X] Changelog entry added.

### CODE REVIEW

File: directory_client_core/helpers.py [hunk 1]
```diff
@@ -146,3 +146,32 @@ def wrapper(client, url, params={}, *args, **kwargs):
...
+    logger.filters = []
+    logger.addFilter(log_filter)
+
+    def closure(func):
+        @wraps(func)
+        def wrapper(client, source_url, *args, **kwargs):
+            cache_key = canonicalize_url(source_url)
+            cached_content = cache.get(cache_key)
+            if cached_content:
+                return cached_content
```
Comment (richtier) [inline directory_client_core/helpers.py]: `return CacheResponse.from_cached_content(cached_content)`

File: directory_client_core/helpers.py [hunk 2]
```diff
@@ -146,3 +146,32 @@ def wrapper(client, url, params={}, *args, **kwargs):
...
     return closure
+
+def ttl(cache, seconds):
+    log_filter = ThrottlingFilter(cache=cache)
+    logger.filters = []
+    logger.addFilter(log_filter)
+
+    def closure(func):
+        @wraps(func)
+        def wrapper(client, source_url, *args, **kwargs):
```
Comment (richtier) [inline directory_client_core/helpers.py]: `url` instead of `source_url` would be more consistent

File: directory_client_core/helpers.py [hunk 3]
```diff
@@ -146,3 +146,32 @@ def wrapper(client, url, params={}, *args, **kwargs):
...
+            if cached_content:
+                return cached_content
+            else:
+                response = func(
+                    client,
+                    source_url=source_url,
+                    *args,
+                    **kwargs,
+                )
+                cache.set(
```
Comment (richtier) [inline directory_client_core/helpers.py]: do we want to check status code? currently, the result of a 5xx will be cached for 30 mins :)

**Comment by agiamas** [comment]
won't implement at this time. Will implement in a simpler single tier architecture.

**Pull request closed without merging**